internal/configs/version1/nginx-plus.tmpl

```diff
@@ -27,9 +27,7 @@ load_module modules/ngx_fips_check_module.so;
 {{$value}}{{end}}
 {{- end}}
 
-{{- if .OIDC}}
 load_module modules/ngx_http_js_module.so;
-{{- end}}
```

Comment:
I wonder if this could be considered a breaking change if a user is building their own image without NJS installed 🤔. Maybe we could give a config option for `enableNJS` which we could default to true and fail this and the OIDC policy if it's not enabled?

Reply:
You're right that loading the same module twice will cause an NGINX emerg error. Which would be breaking for that user. However I don't know if I agree that I would consider it a breaking change as it requires a user to be using their own image, but nevertheless I like the idea of the flag, and we would then have an easy way to conditionally add the JS directives in the main config files.

```nginx
 {{- if .NJSEnabled}}
 js_import /etc/nginx/njs/apikey_auth.js;
 js_set $apikey_auth_hash apikey_auth.hash;
```

<img width="955" alt="image" src="https://github.com/nginxinc/kubernetes-ingress/assets/20595836/195da063-1df9-49a9-9e12-967969088c45">

Reply:
we will do this in another PR as there is a lot in this already 
[#5761](https://github.com/nginxinc/kubernetes-ingress/issues/5761)